lens-cube/src/main/java/org/apache/lens/cube/parse/StorageTableResolver.java

```diff
@@ -169,6 +152,25 @@ public void rewriteContext(CubeQueryContext cubeql) throws LensException {
     phase = phase.next();
   }
 
+  /**
+   * Each candidate in the set is a complex candidate. We will evaluate each one to get
+   * all the partitions needed to answer the query.
+   *
+   * @param cubeql
+   */
+  private void resolveStoragePartitions(CubeQueryContext cubeql) throws LensException {
+    Set<Candidate> candidateList = cubeql.getCandidates();
+    for (Candidate candidate : candidateList) {
+      boolean isComplete = true;
+      for (TimeRange range : cubeql.getTimeRanges()) {
+        isComplete &= candidate.evaluateCompleteness(range, failOnPartialData);
+      }
+      if (!isComplete) {
+        // TODO union : Prune this candidate?
```

Comment:
How do we decide the pruning message ? Is it because of missing partitions or because of incomplete data. We need to discuss this and incorporate in our design